pymobiledevice3/cli/developer.py

```diff
@@ -878,7 +880,19 @@ def debugserver_start_server(service_provider: LockdownClient, local_port):
     else:
         service_name = 'com.apple.internal.dt.remote.debugproxy'
 
-    LockdownTcpForwarder(service_provider, local_port, service_name).start()
+    if (local_port):
+        LockdownTcpForwarder(service_provider, local_port, service_name).start()
+    elif Version(service_provider.product_version) >= Version('17.0'):
+        try:
+            remote_connection = service_provider.start_lockdown_developer_service(service_name).socket
```

Comment:
- this will create an unnecessary tcp handshake
- lets make `_get_service_port()` public (remove the `_`) and use to to get the port number instead

Reply:
Ok. Tad confused here. We still need to actually start the debug server right? This command was essentially pulled from the tcp_forwarder.py code. Is it not required there either? What am I missing? 100% on board if it is somehow already started and we just pull the port number. Should we attempt that and only do start_lockdown_developer_service if the port is non-existent or service isn't started?

I do think get_service_port should be public and have made that change... just want to be sure here. Also of note I need the IP address as well as the port so I can print a proper connection string. (Although that'll be identical to the IP provided for the RSD so that can be re-used here)

Reply:
My code here is wrong regardless since I'm outputting the local IP/port and that's not at all what I want. I'm submitting a change that will just use get_service_port().